client/src/canvas/components/Canvas.tsx

```diff
@@ -182,27 +188,35 @@ class Canvas extends Component<Props, State> {
   }
 
   getMousePos(canvas, evt) {
-    var rect = canvas.getBoundingClientRect();
+    const rect = canvas.getBoundingClientRect();
     return {
       x: evt.clientX - rect.left,
       y: evt.clientY - rect.top
     };
   }
 
-  onCancel(shouldUpdateColor) {
-    // In some states cancel is called without the color picker present (like zooming). 
-    // In this case, we shouldn't update the canvas' color since it would have never changed. 
-    if (!shouldUpdateColor) {
-      this.hideColorPicker(false); 
-    } else {
-      this.hideColorPicker(true);
+  onCancel() {
+    // In some states cancel is called without the color picker present (like zooming).
+    // In this case, we shouldn't update the canvas' color since it would have never changed.
```

Comment:
Lol, what I initially did definitely would have fixed this.... glad to see that you were able to find the bug in your solution though ☺️